### ISSUE TITLE
Add support for MetroCluster software updates

### ISSUE THREAD

**Issue by vrd83** [ansible-collections/netapp.ontap#17]
### Summary

The na_ontap_software_update module is intended for use against a single cluster with few exceptions. Customers with a large number of MetroClusters are unable to use the module to automate ONTAP software updates with Ansible. The na_ontap_software_update module should be updated to support MetroCluster updates, or perhaps a dedicated module to handle MetroCluster updates could be created.

The updated module should handle MetroCluster specific operations, for example copy the updated image files to both clusters in the MetroCluster configuration and run the cluster image validation prior to starting the software update.

In the lab, we were able to update a MCC-FC MetroCluster from 9.9.0 to 9.9.1 using a playbook with a couple work arounds (see 'Additional Information'), however an update from 9.9.1 to 9.10 failed during the validation task even when 'ignore_validation_warning' is set to 'True'.

### Component Name

na_ontap_software_update

### Additional Information

```yaml (paste below)
####################
# Perform upgrades #
####################
---
- hosts: localhost
  name: Update MetroCluster
  gather_facts: false
  collections:
  - netapp.ontap
 
  vars_files:
  - mcc-vars.yaml
  tasks:
 
  - name: "Copy {{ package_name }} package to {{ SiteA_C1_hostname }}"
    na_ontap_software_update:
      state: present
      nodes: 
      - "{{ SiteA_N1_nodename }}"
      - "{{ SiteA_N2_nodename }}"
      package_url: "{{ url }}"
      package_version: "{{ package_name }}"
      https: True
      ignore_validation_warning: False
      validate_certs: False
      download_only: True
      hostname: "{{ SiteA_C1_hostname }}"
      username: "{{ username }}"
      password: "{{ password }}"
      stabilize_minutes: 5

  - name: "Copy {{ package_name }} package to {{ SiteB_C2_hostname }}"
    na_ontap_software_update:
      state: present
      nodes: 
      - "{{ SiteB_N1_nodename }}"
      - "{{ SiteB_N2_nodename }}"
      package_url: "{{ url }}"
      package_version: "{{ package_name }}"
      https: True
      ignore_validation_warning: False
      validate_certs: False
      download_only: True
      hostname: "{{ SiteB_C2_hostname }}"
      username: "{{ username }}"
      password: "{{ password }}"
      stabilize_minutes: 5

  - name: "Start image validation on {{ SiteA_C1_hostname }}"
    na_ontap_command:
      https: True
      validate_certs: False
      hostname: "{{ SiteA_C1_hostname }}"
      username: "{{ username }}"
      password: "{{ password }}"
      command: ['cluster', 'image', 'validate', '-version', "{{ package_name }}"]

  - name: "Start image validation on {{ SiteB_C2_hostname }}"
    na_ontap_command:
      https: True
      validate_certs: False
      hostname: "{{ SiteB_C2_hostname }}"
      username: "{{ username }}"
      password: "{{ password }}"
      command: ['cluster', 'image', 'validate', '-version', "{{ package_name }}"]

  - name: "Wait for image validation to complete"
    wait_for:
      timeout: 60
    delegate_to: localhost

  - name: "Start ONTAP upgrade"
    na_ontap_software_update:
      state: present
      nodes: 
      - "{{ SiteA_N1_nodename }}"
      - "{{ SiteA_N2_nodename }}"
      package_url: "{{ url }}"
      package_version: "{{ package_name }}"
      https: True
      ignore_validation_warning: True
      validate_certs: False
      download_only: False
      force_update: False
      hostname: "{{ SiteA_C1_hostname }}"
      username: "{{ username }}"
      password: "{{ password }}"
      stabilize_minutes: 5
      timeout: 3600
```

**Comment by lonico** [comment]
Looks like the right way to do it currently.   I agree that this could be simplified and we should avoid resorting to na_ontap_command to run validation.

>> however an update from 9.9.1 to 9.10 failed during the validation task 
What was the issue?   Do you have a case for this?   Even if we include the image validation step in the module, it would not be different from the steps you have in the playbook.

**Comment by airborneric** [comment]
however an update from 9.9.1 to 9.10 failed during the validation task
What was the issue? Do you have a case for this? Even if we include the image validation step in the module, it would not be different from the steps you have in the playbook.


TASK [../../../roles/ONTAP : fail] *********************************************18:30:30
49
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Upgrade version is lower than Current ONTAP version"}


This si the issue I am having with 9.8 to 9.10.  My hunch is that in computer logic, 9.10 is lower than 9.2, 9.3, 9.4 etc...

**Comment by lonico** [comment]
Re the issue reported on 1/31.
The validation is done by ONTAP, and they extensively test upgrade scenarios, so I'm doubtful they would have missed this.
This is being followed up on our slack channel.  I'd like to have a full trace to make sure the latest version of the module is used.

**Comment by carchi8py** [comment]
We have an internal story DEVOPS-4870 open for this.

**Comment by lonico** [comment]
We added the following option in November 2021 (collection version 21.11.0)
```
  validate_after_download:
    description:
      - By default validation is not run after download, as it is already done in the update step.
      - This option is useful when using C(download_only), for instance when updating a MetroCluster system.
    default: False
    type: bool
    version_added: 21.11.0
```
And this is improved in 21,22,0 (August 2022) to work with ZAPI and REST.
Would that meet the requirements?

And even though it says after download, if the image is already present, it will run validations, and only validations is download_only is set to true..

**Comment by lonico** [comment]
fixed in 21.22.0
